### DISCUSSION TITLE
How to create aliases for associations

### DISCUSSION THREAD

**Discussion by valentinfp** [mongodb/mongoid#5593] [Q&A]
Let's say I'm working a generic class like

```ruby
  class Test
    include Mongoid::Document
    
    field :title, String
    
    has_many :test_questions
  end
  ```
What's the best way I can make an alias named `questions` for that `test_questions` association? I first tried using `alias_attribute`, but it doesn't work for associations because it tries to create alias for methods that don't exist, like `test_questions_change`, and raises an error.

**Comment by alexbevi** [comment]
Hi @valentinfp, you could do this using a [custom association name](https://www.mongodb.com/docs/mongoid/current/reference/associations/#custom-association-names) similar to the following:

```ruby
class TestQuestion
  include Mongoid::Document
  
  field :title, type: String
  
  belongs_to :test
end

class Test
  include Mongoid::Document
  
  field :title, type: String
  
  has_many :questions, class_name: "TestQuestion"
end
```
